boards/arm/nucleo_f030r8/board.cmake

```diff
@@ -1 +1,8 @@
-include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+set_ifndef(STLINK_FW stlink)
```

Comment:
Do we really need to keep adding these sorts of cmake variables?

With west you can now run `west flash -r jlink` for example.

We could allow setting runners on a per board basis as a west configuration option to make a choice permanent.

Reply:
if it is possible to continue to support Zephyr workflow without specific to Zephyr tools only then why not? The less specific tools are involved in the workflow the better integration with existing IDE's (Eclipse, CLion, VS and etc..) and editor plugins.

Reply:
I'm not sure what you mean by "specific to Zephyr tools". If you're using Zephyr's flashing tools, then you're using west; the build system calls west to flahs. My comment is about how to use west to get this job done.

If you're calling `ninja flash` and relying on the build system to call west instead of running `west flash` directly, you can just add `-DZEPHYR_BOARD_FLASH_RUNNER=jlink`  to your cmake command line to change the default runner instead of `-DSTLINK_FW=jlink`. STLINK_FW is not needed at all as a separate variable -- you can set the default flash runner (and debug, using `ZEPHYR_BOARD_DEBUG_RUNNER`) with existing CMake variables that work for all boards already, without extra boilerplate in each and every board.cmake.

The changes which add jlink-specific runner arguments to the various boards seem fine and can stay. They will only apply if the jlink runner is used. So I'd be fine with this pull request being reworked to just add the `board_runner_args(jlink ...)` bits and removing all the STLINK_FW additions from boards that don't already have that.

Reply:
Reworked, left jlink args and includes for openocd, and jlink. As openocd include is the first one the default runners will be set to openocd.